ambry-account/src/main/java/com/github/ambry/account/HelixAccountService.java

```diff
@@ -133,11 +142,22 @@
     if (notifier != null) {
       notifier.subscribe(ACCOUNT_METADATA_CHANGE_TOPIC, listener);
     }
-    try {
-      readFullAccountAndUpdateCache(FULL_ACCOUNT_METADATA_PATH);
-    } catch (Exception e) {
-      logger.error("Exception occurred when fetching remote account data", e);
-      accountServiceMetrics.fetchRemoteAccountErrorCount.inc();
+    Runnable updater = () -> {
+      try {
+        readFullAccountAndUpdateCache(FULL_ACCOUNT_METADATA_PATH, false);
+      } catch (Exception e) {
+        logger.error("Exception occurred when fetching remote account data", e);
+        accountServiceMetrics.fetchRemoteAccountErrorCount.inc();
+      }
+    };
+    updater.run();
+    if (storeConfig.accountServicePollingIntervalMs > 0) {
+      int initialDelay = new Random().nextInt(storeConfig.accountServicePollingIntervalMs + 1);
```

Comment:
lets not create the scheduler if its not required - infact, if it is possible, lets get the factory to pass this in.

Reply:
Changed as suggested.